Tasks/ContainerBuildV0/src/utils.ts

```diff
@@ -0,0 +1,155 @@
+
+const buildctlToolName = "buildctl"
+const uuidV4 = require('uuid/v4');
+const buildctlLatestReleaseUrl = "https://api.github.com/repos/moby/buildkit/releases/latest";
+const buildctlToolNameWithExtension = buildctlToolName + getExecutableExtension();
+const stableBuildctlVersion = "v0.5.1"
+var serviceName = "k8s-poolprovider"
+var namespace = "azuredevops"
+var port = "8082"
+var clusteruri = ""
+
+export async function getStableBuildctlVersion(): Promise<string> {
+    var request = new webclient.WebRequest();
+    request.uri = buildctlLatestReleaseUrl;
+    request.method = "GET";
+
+    try {
+        var response = await webclient.sendRequest(request);
+        return response.body["tag_name"];
+    } catch (error) {
+        tl.warning(tl.loc("BuildctlLatestNotKnown", buildctlLatestReleaseUrl, error, stableBuildctlVersion));
+    }
+
+    return stableBuildctlVersion;
+}
+
+export async function downloadBuildctl(version: string): Promise<string> {
+
+    let buildctlDownloadPath: string = null;
+    var cachedToolpath = toolLib.findLocalTool(buildctlToolName, version);
+    
+    if (!cachedToolpath) {
+        try {
+            buildctlDownloadPath = await toolLib.downloadTool(getBuildctlDownloadURL(version), buildctlToolName + "-" + version + "-" + uuidV4() + getArchiveExtension());
+        } catch (exception) {
+            throw new Error(tl.loc("BuildctlDownloadFailed", getBuildctlDownloadURL(version), exception));
+        }
+
+        var unzipedBuildctlPath = await toolLib.extractTar(buildctlDownloadPath);
+        unzipedBuildctlPath = path.join(unzipedBuildctlPath, "bin", buildctlToolNameWithExtension);
+        
+        tl.debug('Extracting archive: ' + unzipedBuildctlPath+' download path: '+buildctlDownloadPath);
+        
+        var cachedToolpath = await toolLib.cacheFile(unzipedBuildctlPath, buildctlToolNameWithExtension, buildctlToolName, version);
+        
+        tl.debug('CachedTool path: ' + cachedToolpath);    
+    }
+
+    var buildctlpath = findBuildctl(cachedToolpath);
+    if (!buildctlpath) {
+        throw new Error(tl.loc("BuildctlNotFoundInFolder", cachedToolpath))
+    }
+    
+    tl.debug('Buildctl path: ' + buildctlpath);
+    
+    fs.chmodSync(buildctlpath, "777");
+    return buildctlpath;
+}
+
+function getBuildctlDownloadURL(version: string): string {
+    switch (os.type()) {
+        case 'Windows_NT':
+            return util.format("https://github.com/moby/buildkit/releases/download/%s/buildkit-%s.windows-amd64.tar.gz", version, version);
+
+        case 'Darwin':
+            return util.format("https://github.com/moby/buildkit/releases/download/%s/buildkit-%s.darwin-amd64.tar.gz", version, version);
+
+        default:
+            case 'Linux':
+                return util.format("https://github.com/moby/buildkit/releases/download/%s/buildkit-%s.linux-amd64.tar.gz", version, version);
+
+    }
+}
+
+export async function getServiceDetails() {
```

Comment:
Is this required? Just use poolService input for this. Set it's default value as azure-pipelines-pool:8080. We need to make this default value in the Helm chart or operator as well. If user overrides this in the Helm chart he can do the same on the task side using the poolService input

Reply:
Will set the default value as azure-pipelines-pool but will need the function to fetch the details like cluster uri, namespace and port